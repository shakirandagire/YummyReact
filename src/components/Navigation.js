import React from 'react';
import { Link } from 'react-router-dom';
import Logout from './logout';


const Navigation = () => (

  <nav className="navbar navbar-expand-lg navbar-light bg-light">
    <a className="navbar-brand"><Link to="/main">Yummy Recipes</Link></a>
    <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
      <span className="navbar-toggler-icon" />
    </button>
    <div className="collapse navbar-collapse title" id="navbarNavAltMarkup">
      <div className="navbar-nav">
        <a className="nav-item nav-link active"><Link to="/main">Home </Link><span className="sr-only">(current)</span></a>
        <a className="nav-item nav-link"><Link to="/viewcategories"> Categories </Link></a>
        <a className="nav-item nav-link">Pricing</a>
      </div>
      <Logout />
    </div>
  </nav>

);

export default Navigation;
