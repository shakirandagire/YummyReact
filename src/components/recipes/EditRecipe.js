import React, { Component } from 'react';
import axios from 'axios';
import { notify } from 'react-notify-toast';
import { BASE_URL } from '../../../src/constants';
import Navigation from '../Navigation';

/**
 * Component for editing recipes
 */

class EditRecipe extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipename: '',
      recipe_description: '',
      instructions: '',
    };
  }
  // Function that populates the edit recipes form
  componentWillMount() {
    const headers = { Authorization: `Bearer ${localStorage.getItem('accessToken')}` };
    const catId = this.props.match.params.category_id;
    const recipeId = this.props.match.params.id;
    axios
      .get(`${BASE_URL}/api/v1/categories/${catId}/recipes/${recipeId}`, { headers })
      .then((response) => {
        this.setState({
          recipename: response.data.recipename,
          recipe_description: response.data.recipe_description,
          instructions: response.data.instructions,
        });
      })
      .catch((error) => {
        if (error.response) {
          notify.show(error.response.data.message);
        } else if (error.request) {
          notify.show("Request not made");
        }
      });
  }

  handleInputChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }
  // Function that handles the editting of recipes

  handleEditRecipe = (event) => {
    const headers = { Authorization: `Bearer ${localStorage.getItem('accessToken')}` };
    event.preventDefault();
    const catId = this.props.match.params.category_id;
    const recipeId = this.props.match.params.id;
    axios
      .put(`${BASE_URL}/api/v1/categories/${catId}/recipes/${recipeId}`, this.state, { headers })
      .then(() => {
        notify.show("Recipe edited successfully", 'success', 4000);
        this.props.history.push(`/view-recipes/${catId}/recipes`);
      })
      .catch((error) => {
        if (error.response) {
          notify.show(error.response.data.message, 'success', 4000);
        } else if (error.request) {
          notify.show("Request not made");
        }
      });
  }

  render() {
    return (
      <div className="my_container backgimg">
        <Navigation />
        <div className="mycontent">

          <form onSubmit={this.handleEditRecipe}>

            <div className="form-group">
              Recipe Name:
              <input type="text" name="recipename" className="form-control" required onChange={this.handleInputChange} value={this.state.recipename} />
            </div>
            <div className="form-group">
              Description:
              <input type="text-area" name="recipe_description" className="form-control" required onChange={this.handleInputChange} value={this.state.recipe_description} />
            </div>
            <div className="form-group">
              Instructions:
              <input type="text-area" name="instructions" className="form-control" required onChange={this.handleInputChange} value={this.state.instructions} />
            </div>

            <button type="submit" className="btn btn-danger formsave" >Save</button>
          </form>
        </div>

      </div>


    );
  }
}
export default EditRecipe;
