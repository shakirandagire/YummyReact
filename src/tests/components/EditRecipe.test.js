import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import EditRecipe from '../../components/recipes/EditRecipe';
import Navigation from '../../components/other/Navigation';

describe('<EditRecipe/>', () => {
  const params = {
    match: {
      params: {
        id: 1,
      },
    },
  };
  const preventDefault = jest.fn();
  const component = shallow(<EditRecipe match={{ params }} />);
  it('should render properly', () => {
    expect(component.length).toBe(1);
  });

  it('should render edit recipes', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('should have edit recipe method', () => {
    expect(component.instance().handleEditRecipe({ preventDefault }));
  });
  it('should have initial state', () => {
    expect(component.state().recipename).toEqual("");
    expect(component.state().recipe_description).toEqual("");
    expect(component.state().instructions).toEqual("");
  });
  it('should change state', () => {
    component.setState({ recipename: 'brunch', recipe_description: 'breakfast and lunch',instructions: 'Sauages and eggs' });
    expect(component.find('[name="recipename"]').props().value).toEqual('brunch');
    expect(component.find('[name="recipe_description"]').props().value).toEqual('breakfast and lunch');
    expect(component.find('[name="instructions"]').props().value).toEqual('Sauages and eggs');
  });
  it('renders a form', () => {
    expect(component.find('form')).toHaveLength(1);
    expect(component.find('form').simulate('submit', { preventDefault }));
    expect(preventDefault).toBeCalled();
  });
  it('should render <Navigation /> component', () => {
    const wrapper = shallow(<Navigation />);
    expect(wrapper.length).toEqual(1);
  });
});
