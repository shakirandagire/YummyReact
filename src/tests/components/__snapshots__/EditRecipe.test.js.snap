// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditRecipe/> should render edit recipes 1`] = `
<div
  className="my_container backgimg"
>
  <Navigation />
  <div
    className="mycontent"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        Recipe Name:
        <input
          className="form-control"
          name="recipename"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        Description:
        <input
          className="form-control"
          name="recipe_description"
          onChange={[Function]}
          required={true}
          type="text-area"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        Instructions:
        <input
          className="form-control"
          name="instructions"
          onChange={[Function]}
          required={true}
          type="text-area"
          value=""
        />
      </div>
      <button
        className="btn btn-danger formsave"
        type="submit"
      >
        Save
      </button>
    </form>
  </div>
</div>
`;
